bundles/specmate-persistency-cdo/src/com/specmate/persistency/validation/LibraryFolderValidator.java

```diff
@@ -0,0 +1,55 @@
+package com.specmate.persistency.validation;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.EStructuralFeature;
+
+import com.specmate.common.SpecmateValidationException;
+import com.specmate.model.support.util.SpecmateEcoreUtil;
+import com.specmate.persistency.IChangeListener;
+import com.specmate.persistency.event.EChangeKind;
+
+public class LibraryFolderValidator implements IChangeListener {
+
+	// NOT NEEDED?
+
+	@Override
+	public void changedObject(EObject object, EStructuralFeature feature, EChangeKind changeKind, Object oldValue,
+			Object newValue, String objectClassName) throws SpecmateValidationException {
+
```

Comment:
shouldn't valdidateNotTopLevelLibraryFolder be called? If not, this class is in fact not needed

Reply:
I forgot to delete this class. The validation is already covered by TopLevelValidator since it is not necessary to distinguish between project top-level or library top-level folder; both have the same restrictions, are only applied in different contexts (which are controllable by when the validator is added to the transaction).

Note that TopLevelValidator is currently not used as it completely breaks the integration tests, which were not written with that constraint in mind. Adapting the tests is a major undertaking. I added a task to Trello, but I would not wait to merge this code because of that.

Reply:
I agree